### PR TITLE
[New Designs] Update Chose websites view

### PR DESCRIPTION
**Pull request by aanorbel**
Fixes https://github.com/ooni/probe/issues/2592

## Proposed Changes

  - Updated `CustomWebsiteActivity` to use new theme.
  - Change List item builder to use  `RecyclerView`.

Tasks

- [x] Implement designs on light theme.
- [x] Convert list item builder to  use  `RecyclerView`.
- [x] Fix dark theme
- [x] Implement modal properly

| Light | Dark|
|-|-|
|Screenshot_20231104_110235)| Screenshot_20231104_110254 |
|Screenshot_20231104_110336| Screenshot_20231104_110407 |
| Screenshot_20231104_110429 | Screenshot_20231104_110441 |

### CODE REVIEW

**Review by bassosimone — CHANGES_REQUESTED** [review]
Mostly 🐳. Maybe we need an issue for the TODO unless we want to address it as part of this pull request?

File: app/src/main/java/org/openobservatory/ooniprobe/activity/customwebsites/CustomWebsiteActivity.kt [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+import org.openobservatory.ooniprobe.activity.customwebsites.adapter.ItemChangedListener
+import org.openobservatory.ooniprobe.common.PreferenceManager
+import org.openobservatory.ooniprobe.databinding.ActivityCustomwebsiteBinding
+import org.openobservatory.ooniprobe.fragment.ConfirmDialogFragment
+import org.openobservatory.ooniprobe.model.database.Url
+import org.openobservatory.ooniprobe.test.suite.WebsitesSuite
+import java.io.Serializable
+import javax.inject.Inject
+
+class CustomWebsiteActivity : AbstractActivity(), ConfirmDialogFragment.OnClickListener {
```
Comment (bassosimone) [inline app/src/main/java/org/openobservatory/ooniprobe/activity/customwebsites/CustomWebsiteActivity.kt:25]:
```suggestion
/** This activity allows a user to test a custom website. */
class CustomWebsiteActivity : AbstractActivity(), ConfirmDialogFragment.OnClickListener {
```

File: app/src/main/java/org/openobservatory/ooniprobe/activity/customwebsites/CustomWebsiteActivity.kt [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+    @Inject
+    lateinit var preferenceManager: PreferenceManager
+
+    val viewModel: CustomWebsiteViewModel by viewModels()
+
+    private lateinit var adapter: CustomWebsiteRecyclerViewAdapter
+    private lateinit var binding: ActivityCustomwebsiteBinding
+    override fun onCreate(savedInstanceState: Bundle?) {
```
Comment (bassosimone) [inline app/src/main/java/org/openobservatory/ooniprobe/activity/customwebsites/CustomWebsiteActivity.kt:33]:
```suggestion

    override fun onCreate(savedInstanceState: Bundle?) {
```

**Review by bassosimone — APPROVED** [review]
🐳 LGTM with minor suggestions

File: app/src/main/java/org/openobservatory/ooniprobe/activity/OverviewActivity.java [hunk 3]
```diff
@@ -11,6 +11,7 @@
 import androidx.core.view.ViewCompat;
 
 import org.openobservatory.ooniprobe.R;
+import org.openobservatory.ooniprobe.activity.customwebsites.CustomWebsiteActivity;
```
[the remark this answers is not in the document]
Comment (aanorbel) [reply app/src/main/java/org/openobservatory/ooniprobe/activity/OverviewActivity.java:14]: Required at L106

**Pull request merged**